### ISSUE TITLE
Getting an error as null is not an object (evaluating '_nativeInterface.default.getPhotos)

### ISSUE THREAD

**Issue by nihp** [react-native-community/react-native-cameraroll#113]
I am getting an error while trying to open Cameraroll (getPhotos)

**Comment by Prateek93a** [comment]
yes I am getting the same error

**Comment by Jarred-Sumner** [comment]
Are you sure that you ran `pod install`?

**Comment by nihp** [comment]
It working fine for me now

**Comment by Dagurmart** [comment]
I get this error after upgrading Expo to version 36

**Comment by lucastschmidt** [comment]
Me too ... same error =/ after upgrading version 36

**Comment by Dagurmart** [comment]
> Me too ... same error =/ after upgrading version 36

I stopped using this package and switched to [MediaLibrary](https://docs.expo.io/versions/latest/sdk/media-library/) so now I do
      
```
const album = await MediaLibrary.getAlbumAsync('Camera')
const photosTemp = await MediaLibrary.getAssetsAsync({ album: album })
```

**Comment by lucastschmidt** [comment]
Expo removed the package! I will post below the code change we had to do:

Migrate to MediaLibrary. Be aware, the return is different than the CameraRoll. 

```  async getImages(params: Object) {
    const {
      assets: images,
      ...pageInfo
    } = await MediaLibrary.getAssetsAsync({
      ...params,
    });
    this.setState({ images });
    return { images, pageInfo };
  }```

```  async getMoreImages() {
    InteractionManager.runAfterInteractions(async () => {
      const { endCursor, hasNextPage, images } = this.state;

      if (!hasNextPage) {
        return;
      }

      const { images: newImages, pageInfo } = await this.getImages({
        first: INITIAL_ITEMS,
        after: endCursor,
      });

      this.setState({
        images: (images || []).concat(newImages),
        endCursor: pageInfo.endCursor,
        hasNextPage: pageInfo.hasNextPage,
      });
    });
  }```

Be aware, it changed a bit the signature of the getPhotos method, not difficult, but the param names and such. If you change from CameraRoll MediaLibrary

**Comment by nihp** [comment]
My getPhotos() method flickering and it did not display all images in the Library of my phone. 

Can anyone have any solution. Same error is reproducible again

**Comment by nihp** [comment]
Fixed by 

import CameraRoll from "@react-native-community/cameraroll";

**Comment by SanjanaTailor** [comment]
It is not working for me after adding same import given there.
same error getPhotos

```import { ActionSheetIOS, ListView, StyleSheet, Text, TouchableOpacity, View, Platform} from 'react-native';
import { Navigator } from 'react-native-deprecated-custom-components';
import PhotoBrowser from 'react-native-photo-browser';
import CameraRoll from "@react-native-community/cameraroll";


const Photos= [
  {
    title: 'A',
    description: 'Photo to display',
    enableGrid: false,
    media: [{
      photo: 'A',
      title: 'A',
    }],
  }
];


CameraRoll.getPhotos({
  first: 30,
  assetType: 'Photos',
}).then((data) => {
  const media = [];
  data.edges.forEach(d => media.push({
    photo: d.node.image.uri,
  }));
  Photos[2].media = media;
}).catch();```

**Comment by izzykreie** [comment]
I'm have the same issue. @SanjanaTailor @nihp what did you guys do to fix this?

**Comment by imanshu1** [comment]
Any updates on this ?

**Comment by izzykreie** [comment]
@imanshu1 had to install the library and cd into my iOS folder and run command "pod install" I never did a pod install for the library.

**Comment by imanshu1** [comment]
@izzykreie what if i don't have pod file any podfile in my project, is there any other option for configuration?

**Comment by izzykreie** [comment]
Are you developing on android? Or iOS? if you're trying to run this project on an iOS device you will have a pod file once you run the command to download the library. if you're trying to run on an android it is a different set of steps that do not requite a pod file. they use a manifest and gradle file in react native. if you are using iOS like I am and need to get that pod file, all you need to do is run the command to download the library, and then do cd iOS pod install. steps are here: https://github.com/react-native-community/react-native-cameraroll

**Comment by vzts** [comment]
I get this in Android. Our service was receiving a lot of this error when user selected nothing but just back pressed.
```
TypeError null is not an object (evaluating 'n.default.getPhotos') 
    node_modules/@react-native-community/cameraroll/js/CameraRoll.js:206:25 value
```
This is a promise so it didn't really make a crash but it does give an alert on our analytics.

I suspect the culprit is below code
```
  static getPhotos(params: GetPhotosParams): Promise<PhotoIdentifiersPage> {
    if (!params.assetType) {
      params.assetType = ASSET_TYPE_OPTIONS.All;
    }
    if (!params.groupTypes && Platform.OS !== 'android') {
      params.groupTypes = GROUP_TYPES_OPTIONS.All;
    }
    if (arguments.length > 1) {
      console.warn(
        'CameraRoll.getPhotos(tag, success, error) is deprecated.  Use the returned Promise instead',
      );
      let successCallback = arguments[1];
      const errorCallback = arguments[2] || (() => {});
      RNCCameraRoll.getPhotos(params).then(successCallback, errorCallback);
    }
    return RNCCameraRoll.getPhotos(params);
  }
```
when `RNCCameraRoll.getPhotos(params);` does not return a proper `PhotoIdentifiersPage`.

**Comment by vzts** [comment]
In my case, I get this error only on Android.
The code I'm using:
```js
const response = await CameraRoll.getPhotos({ first: 1 })
```
The error I'm getting:
```
TypeError null is not an object (evaluating 'n.default.getPhotos')
    node_modules/@react-native-community/cameraroll/js/CameraRoll.js:206:25 value
```
This is a promise so it didn't really make a crash but it does give an alert on our analytics.

I suspect the culprit is below code
```js
  static getPhotos(params: GetPhotosParams): Promise<PhotoIdentifiersPage> {
    if (!params.assetType) {
      params.assetType = ASSET_TYPE_OPTIONS.All;
    }
    if (!params.groupTypes && Platform.OS !== 'android') {
      params.groupTypes = GROUP_TYPES_OPTIONS.All;
    }
    if (arguments.length > 1) {
      console.warn(
        'CameraRoll.getPhotos(tag, success, error) is deprecated.  Use the returned Promise instead',
      );
      let successCallback = arguments[1];
      const errorCallback = arguments[2] || (() => {});
      RNCCameraRoll.getPhotos(params).then(successCallback, errorCallback);
    }
    return RNCCameraRoll.getPhotos(params);
  }
```
RNCCameraRoll.getPhotos(params) does not seem to return a proper PhotoIdentifiersPage.

lib version: 1.5.2

**Comment by vzts** [comment]
Okay... so it turned out that somehow `react-native link @react-native-community/cameraroll` did not add
```java
packages.add(new CameraRollPackage());
```
in MainApplication.java on Android. I thought the linking command would do so, strange.

**Comment by aterribili** [comment]
I used the autolink of react-native 0.60+ and didn't work. Autolink + adding `new CameraRollPackage()` on MainApplication.java `getPackages()` works fine.  Maybe we have some problems on the autolink script.

**Comment by windzzf** [comment]
I also encountered the same problem, how did you solve it？

**Comment by gabrielgb03** [comment]
I have the same problem and i can't solve this, how do you fix?

**Comment by PatrissolJuns** [comment]
@gabrielgb03 You just need to rebuild your app. So run `react-native run-android` or `react-native run-ios`. It worked for me

**Comment by bartolkaruza** [comment]
Hi, this issue has become a bucket for various linking issues. If you use expo, media-library is the logical default option there. This repo and issue list covers the standalone cameraroll library, which was extracted from React Native core. There appears to be an issue with linking on Android, where in some cases the package is not added correctly to the android native code, but I'm not able to reproduce this on my end. If someone can share a minimal repro of a specific case, please open a new issue and post the reproduction repo/steps there.

**Comment by ckswopnera** [comment]
import this => import {CameraRoll} from '@react-native-camera-roll/camera-roll';
instead of => import CameraRoll from '@react-native-camera-roll/camera-roll';
